src/recipe.cpp

```diff
@@ -738,34 +696,95 @@ std::set<proficiency_id> recipe::assist_proficiencies() const
     return ret;
 }
 
-float recipe::proficiency_time_maluses( Character &guy ) const
+static float get_aided_proficiency_level( const Character &crafter, proficiency_id prof )
+{
+    float max_prof = crafter.get_proficiency_level( prof );
+    for( const npc *helper : crafter.get_crafting_helpers() ) {
+        max_prof = std::max( max_prof, helper->get_proficiency_level( prof ) );
+    }
+    return max_prof;
+}
+
+static float proficiency_time_malus( const Character &crafter, const recipe_proficiency &prof )
+{
+    if( !crafter.has_proficiency( prof.id ) &&
+        !helpers_have_proficiencies( crafter, prof.id ) && prof.time_multiplier > 1.0f ) {
+        double malus = prof.time_multiplier - 1.0;
+        malus *= 1.0 - crafter.crafting_inventory().get_book_proficiency_bonuses().time_factor( prof.id );
+        double pl = get_aided_proficiency_level( crafter, prof.id );
+        // Sigmoid function that mitigates 100% of the time malus as pl approaches 1.0
+        // but has little effect at pl < 0.5. See #49198
+        malus *= 1.0 - std::pow( 0.5 + 0.5 * std::sin( pl * M_PI - M_PI / 2 ), 2 );
```

Comment:
You can simplify this

sin( πx - π/2 )
= -sin( π/2 - πx )
= -cos( πx )
```suggestion
        malus *= 1.0 - std::pow( 0.5 - 0.5 * std::cos( pl * M_PI ), 2 );
```
(although FWIW personally I think a convex curve makes more sense; I liked x^4/2)

Reply:
Thanks, done